arc_cache.go

```diff
@@ -42,11 +45,32 @@ func newARCCachedBS(ctx context.Context, bs Blockstore, lruSize int) (*arccache,
 	return c, nil
 }
 
+func (b *arccache) rLock(k cid.Cid) func() {
+	b.lks[k.KeyString()[len(k.KeyString())-1]].RLock()
```

Comment:
`func mutexKey(k cid.Cid) int { ... }`?

Also, these allocate. I'd consider something like `getLock(k cid.Cid) *sync.RWMutex { ... }`.

Reply:
Will go with the getLock method.